Orion: Client Changes in preparation for 1.0.0

ungaralex opened the pull request:
<!-- Thanks for contributing to Artemis! Before you submit your pull request, please make sure to check the following boxes by putting an x in the [ ] (don't: [x ], [ x], do: [x]) -->
<!-- If your pull request is not ready for review yet, create a draft pull request! -->

### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Client: I added multiple integration tests (Jest) related to the features
- [x] Client: I documented the TypeScript code using JSDoc style.
- [x] Client: I added multiple screenshots/screencasts of my UI changes
- [x] Client: I translated all the newly inserted strings into German and English

### Motivation and Context
<!-- Why is this change required? What problem does it solve? -->
<!-- If it fixes an open issue, please link to the issue here. -->
The connector between Orion and the client got too big (on the Orion site). We had to split it up into smaller subconnectors, which cover their own set of functionalities. 
The client was not consistent regarding the naming of components/services and properties. We often had "IntelliJ" in these names, although we should stick to a more generic convention.

### Description
<!-- Describe your changes in detail -->
- I renamed all remaining occurrences of "intellij..." to "orion..." or "ide..."
- I adapted the client interfaces to fit the new Orion architecture
- There also is a new version checker, that compares the required Orion version with the one used in the actual installed plugin. If the version is not in the allowed range, the user gets automatically routed to an error page, that informs him which version he uses and which is supported.
- The displayed test results in IntelliJ are now also reflect the task hierarchy from the exercise instructions, e.g. if you have: Task 1(Subtask 1, Subtask 2), Task2; then you will have the following test result structure: Task1[with test results for Task 1](Subtask 1[with test results for subtask 1]......

### Steps for Testing
<!-- Please describe in detail how the reviewer can test your changes. -->
**Outdated version**: [orion_outdated-0.9.0-rc.1.zip](https://github.com/ls1intum/Artemis/files/4091657/orion_outdated-0.9.0-rc.1.zip)
**Working version**: [orion-1.0.0-rc.2.zip](https://github.com/ls1intum/Artemis/files/4144887/orion-1.0.0-rc.2.zip)



1. Log in to Artemis
2. Import an exercise as a student/instructor
3. Run the tests remotely
4. All results should be properly formatted

Test the outdated version checker:
1. Log in to Artemis with an outdated version of Orion
2. You should see the error page informing you about the required version

### Screenshots
<!-- Add screenshots to demonstrate the changes in the UI. -->
<!-- Create a GIF file from a screen recording in a docker container https://toub.es/2017/09/11/high-quality-gif-with-ffmpeg-and-docker/ -->
Outdated version warning:
![image](https://user-images.githubusercontent.com/27979674/72815972-c43f4400-3c67-11ea-8ddf-89de7a9ba746.png)

Structured test results:
![orion_structured_test_results](https://user-images.githubusercontent.com/27979674/72816047-e0db7c00-3c67-11ea-99cc-efb28608cf1e.gif)

ungaralex commented:
The updating-result and "Scores" button might overlap on some screen size due to a CSS error. You can ignore this for now

madwau commented:
> The updating-result and "Scores" button might overlap on some screen size due to a CSS error. You can ignore this for now

This has been fixed now. It was caused by a Safari (webkit) flexbox bug (see more details [here](https://github.com/philipwalton/flexbugs#flexbug-11)).

ungaralex commented:
I fixed a couple of issues in the plugin and updated the .zips in the description. Please download and re-install them if you want to test something again.

I also added the `resumeExercise` button, to the IDE actions component, which fixes https://github.com/ls1intum/Orion/issues/16
